### PR TITLE
LPS-71728 delegate from existing plugin class to two new classes

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

**Comment by gamerson** [comment]
ci:reopen

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java [hunk 1]
```diff
@@ -14,44 +14,32 @@
...
-		analyzer.setProperty(
-			Constants.PROVIDE_CAPABILITY, provideCapabilityHeaders.toString());
-
-		return true;
+		return modified;
 	}
 
+	private final AnalyzerPlugin[] _delegateAnalyzerPlugins =
+		new AnalyzerPlugin[] {
+			new ProvidesResourceBundleLoaderAnalyzerPlugin(),
```
Comment (Ithildir) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java]: Can I sort these?

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		Parameters provideCapabilityHeaders = new SortedParameters(
+			analyzer.getProperty(Constants.PROVIDE_CAPABILITY));
+
+		Parameters requireCapabilityHeaders = new SortedParameters(
+			analyzer.getProperty(Constants.REQUIRE_CAPABILITY));
+
+		List<String> aggregateKeys = new ArrayList<>(
+			aggregateResourceBundlesParameters.keySet());
+
+		Collections.sort(aggregateKeys);
```
Comment (Ithildir) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java]: Do we really have to sort again? Doesn't `SortedParameters` already do it?

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		if (servletContextName != null) {
+			try {
+				Path servletContextPath = Paths.get(servletContextName);
+
+				servletContextName = servletContextPath.subpath(
+					0, 1).toString();
+			}
+			catch (Exception e) {
+
+				// ignore
```
Comment (Ithildir) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java]: What kind of exception are we ignoring here?

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java (hunk 2 above)
Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java]: Yes i believe we can remove this line.  That was in there from before I created the SortedParameters class.  This should be good to remove.

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java (hunk 3 above)
Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java]: Its a Paths.get() exception.  So if the `Web-Contextpath` can't get parsed by `Paths.get()` then it just doesn't correctly encode the servletContextName.  I guess we shoudl fail the plugin or fail the build maybe?  So the developer knows to fix their `Web-ContextPath`?

File: modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java (hunk 1 above)
Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java]: Yes I don't think order matters

Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java]: But could you send this back to me so I can test it out with blade samples?  To make sure things are still working?

Comment (gamerson) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java]: Or I could send you a branch on blade samples where you can test it out yourself.

Comment (Ithildir) [inline modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/ResourceBundleLoaderAnalyzerPlugin.java]: I'll send it back to you!

**Comment by Ithildir** [comment]
Sent to https://github.com/brianchandotcom/liferay-portal/pull/48459.

**Pull request closed without merging**